subsys/bluetooth/host/mesh/proxy.c

```diff
@@ -473,6 +493,7 @@ static ssize_t proxy_recv(struct bt_conn *conn,
 			return -EINVAL;
 		}
 
+		k_delayed_work_submit(&client->sar_timer, PROXY_SAR_TIMEOUT);
```

Comment:
> I think the timer should be rescheduled for each SAR_CONT

@joerchan yes, that sounds reasonable - the Mesh spec is pretty vague regarding this. The current version of the patch is almost a direct backport of https://github.com/apache/mynewt-nimble/pull/457 so if this needs fixing (which I think it does) then nimble should also be fixed.

@michal-narajowski @rymanluk any comments?

Reply:
I thought about this. "The timeout for the SAR transfer is 20 seconds." I understand it as 20 s for the full transfer from start of segmentation to end of reassembly. But it is not a strong opinion.

Reply:
Usually (and I admit this is a fairly subjective view) this kind of timers are there for catching inactivity, i.e. if *nothing* happens for awhile. A SAR_CONT counts as *something* so in that sense it would be logical for it to reset the timer.

Reply:
I think as long as there is progress, the timeout should be reset.
In the (unlikely) case where you have a 2 second connection interval, and only send 1 segment per connection interval you would very limited in the number of segments you could support.

Reply:
@joerchan sounds like we are in agreement then. I've pushed an updated version which resets the timer also for SAR_CONT.